Issue4〜6の解消

daigo-tkhs opened the pull request:
## 動作確認用Gyazo GIF
予定を入力し「保存」ボタンを押すとデータが保存され、日付のボックスに表示されることを確認しました。  
https://gyazo.com/761596970b8896e83d61a96cb9972617

## WHAT
- Issue4〜6を解消しました
  - 保存処理の修正
  - 日付が表示されないバグの修正
  - 曜日の表示を追加

## WHY
- 予定が正しく保存・表示されるようにするため
- カレンダー表示に必要な日付・曜日情報を正しく出力するため

daigo-tkhs committed:
BundleInstall実行

daigo-tkhs committed:
Issue4

## WHAT
- `plan_params` メソッドのキー指定を修正
- `params.require(:calendars)` → `params.require(:plan)` に変更

## WHY
- フォームから送られてくるパラメータは `"plan" => { "date" => ..., "plan" => ... }` という形になっている
- 元のままだと、`params.require(:calendars)` は存在しないキーを参照しており、`ParameterMissing` エラーが発生していた
- 正しいキー `plan` を使うことで、受け取った値を安全に抽出でき、予定を保存できるようになる

daigo-tkhs committed:
Issue5

## WHAT
- `index.html.erb` の日付部分の出力方法を修正
- `<% day[:month] %>/<% day[:date] %>` → `<%= day[:month] %>/<%= day[:date] %>`

## WHY
- `<% %>` は処理のみ実行し出力しないため、日付が画面に表示されなかった
- `<%= %>` を使うことで、計算結果をビューに正しく出力できる
- 修正により、トップページに「月/日」の形式で日付が表示されるようになった

daigo-tkhs committed:
Issue6

## WHAT
- `calendars_controller.rb` の `getWeek` メソッドに曜日情報を追加
- `@week_days` に `:wday` を含めるよう修正
- `index.html.erb` にて日付に曜日を表示するよう修正

## WHY
- カレンダー表示に曜日の情報を追加する必要があった
- Rubyの `Date#wday` で曜日番号を取得し、`wdays` 配列から対応する文字列を参照
- これにより、カレンダーに「月/日(曜日)」の形式で日付を表示できるようになった

daigo-tkhs committed:
Merge branch 'master' into second_training

OhyaYouki reviewed (commented):
レビュー依頼確認しました。

こちらのプルリクエストですが、
コード内容と合わせて実際の画面や機能ができているか確認させていただきたいので、
先に以下のGyazo動画をご提出いただいてもよろしいでしょうか？

```
予定を入力し「保存」ボタンを押すとデータが保存され、日付のボックスに表示される。
```

Gyazo GIFで生成されたURLをこちらのプルリクエスト内に添付いただけますと幸いです。

その後、再度レビュー依頼フォームにてプルリクエストURLのご提出いただけますでしょうか。
ご対応のほど、よろしくお願いいたします。

OhyaYouki commented on app/controllers/calendars_controller.rb:
```diff
@@ -34,7 +34,12 @@ def get_week
+      # ← ここで曜日を算出
+      wday_num = (@todays_date + x).wday
+      
+      days = { :month => (@todays_date + x).month, :date => (@todays_date+x).day, :plans => today_plans,wday: wdays[wday_num]}# ← 追加！
```
```rb
      days = { :month => (@todays_date + x).month, :date => (@todays_date+x).day, :plans => today_plans,wday: wdays[wday_num]}# ← 追加！
```
こちらはハッシュロケットではなく、シンボル型で記述しましょう。

OhyaYouki commented on app/views/calendars/index.html.erb:
```diff
@@ -1,16 +1,26 @@
-      <div class='date'>
-        <% day[:month] %>/<% day[:date] %>
+<div>
+  <%= form_with model: @plan, url: calendars_path, class: 'form' do |f| %>
```
不要なdivタグを削除しましょう。

daigo-tkhs replied:
対応完了いたしました。
再度レビューを依頼いたします。

OhyaYouki reviewed (commented):
レビュー依頼確認しました。

コメントいたしましたので、ご確認をお願いいたします。

コードの修正後、コミット・プッシュを行った上で再度レビュー依頼をお送りください。
よろしくお願いいたします。

daigo-tkhs committed:
Fix: コントローラーのハッシュ表記とビューの不要タグを修正

## WHAT
- `calendars_controller.rb` のハッシュ表記をシンボル型に統一
- `index.html.erb` の不要なdivタグを削除

## WHY
- Rubyの命名規則に従い、コードの可読性と一貫性を保つため
- 無駄なタグを削除することでビューの構造をシンプルにし、可読性・保守性を向上させるため

daigo-tkhs committed:
Conflictの解消

daigo-tkhs committed:
Conflictの解消

OhyaYouki reviewed (commented):
レビュー依頼確認しました。

**LGTM**とさせていただきます！

これでissue1~6全てのissueが解消されました。お疲れ様でした！
忘れずにこのブランチをMergeしていただき、ローカルのMasterにPullをお願いいたします。

次のカリキュラムも頑張っていきましょう！応援しております！
